rust/xaynet-analytics/src/database/analytics_event/data_model.rs

```diff
@@ -1,34 +1,49 @@
+use anyhow::{anyhow, Result};
 use chrono::{DateTime, Utc};
-use isar_core::object::{data_type::DataType, object_builder::ObjectBuilder};
-use std::vec::IntoIter;
+use std::convert::{From, Into, TryFrom, TryInto};
 
 use crate::database::{
-    common::{FieldProperty, IsarAdapter, SchemaGenerator},
+    analytics_event::adapter::{AnalyticsEventAdapter, AnalyticsEventRelationalAdapter},
+    common::RelationalField,
     screen_route::data_model::ScreenRoute,
 };
 
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub enum AnalyticsEventType {
     AppEvent = 0,
-    Error = 1,
+    AppError = 1,
     ScreenEnter = 2,
     UserAction = 3,
 }
 
+impl TryFrom<i32> for AnalyticsEventType {
+    type Error = ();
```

Comment:
Here we can use `anyhow::Error` and return something more informative in the default case.

Reply:
i'll do it in the next unit tests PR, if that's ok. i'm also fixing some other stuff

Reply:
actually, i pushed it now to avoid git shenanigans in my local branch